manager_janitor.go

```diff
@@ -335,6 +335,105 @@ func (mgr *Manager) pindexesRestart(
 	return errs
 }
 
+func (mgr *Manager) activatePIndex(pi *pindexRestartReq) (*PIndex, error) {
+	err := mgr.stopPIndex(pi.pindex, false)
+	if err != nil {
+		log.Printf("janitor: error closing pindex %s", pi.pindex.Name)
+		return nil, err
+	}
+	// rename the pindex folder and name as per the new plan
+	newPath := mgr.PIndexPath(pi.planPIndexName)
+	if newPath != pi.pindex.Path {
+		log.Printf("janitor: proceeding to rename %s to %s", pi.pindex.Path, newPath)
+		err := os.Rename(pi.pindex.Path, newPath)
+		if err != nil {
+			cleanDir(pi.pindex.Path)
+			cleanDir(newPath)
+			return nil, fmt.Errorf("janitor: restartPIndex"+
+				" updating pindex: %s path: %s failed, err: %v",
+				pi.pindex.Name, newPath, err)
+		}
+		_, err2 := os.Stat(pi.pindex.Path)
+		if err2 == nil { // old pindex dir still exists.
+			err := os.RemoveAll(pi.pindex.Path) // quick fix - deleting it.
+			if err != nil {
+				return nil, fmt.Errorf("janitor: error removing path %s: %s",
+					pi.pindex.Path, err.Error())
+			}
+		}
+		_, err2 = os.Stat(newPath)
+		if os.IsNotExist(err2) {
+			return nil, fmt.Errorf("janitor: error renaming to %s: %s",
+				newPath, err.Error())
+		}
+	}
+	pi2 := pi.pindex.Clone() //pi2 is the new pindex.
+	pi2.Name = pi.planPIndexName
+	pi2.Path = newPath
+
+	_ = mgr.unregisterPIndex(pi.pindex.Name, pi.pindex)
```

Comment:
stoppindex in the top already does this.
So, please revisit all these API's usage judiciously.